internal/target/vmware_target.go

```diff
@@ -0,0 +1,17 @@
+package target
+
+type VMWareTargetOptions struct {
+	Filename   string `json:"filename"`
```

Comment:
While I agree on using ImageName for the filename in vSphere, it seems redundant to pass it as an option since it's already part of the `Target`, unless I'm missing some other use for the options.

PS: considering the optimization discussed above, I don't think `Filename` option is needed either

Reply:
`Filename` is also passed to the job as `ImageName` arg, resulting in 2 ImageNames with different meanings. That might be useful to clean up in separate PR

Reply:
The API needs some more documentation to make this clear. I got myself confused by it yesterday (luckily @larskarlitski saved me from making nonsense comments on this PR).

`ImageName` in the target is indeed the name we want to give to the file/image/artifact in the target cloud, so it makes sense to use that. From your demo it seems that in vSphere there is also a directory, which the image ends up in, and this I thought it may makes sense to make configurable?

`ImageName` at the top level has a different meaning (and really should be documented). When it is set, it is the name of the image osbuild produces which should be uplodad to composer. This makes sense in the weldr usecase, but in general we want to be able to (in the future, from the cloudAPI) to only upload to vSphere, and not back to composer. We would do this by leaving `ImageName` unset.

We want to make a new version of this API, but for now I think we should just document the current behavior and not change it (rather replace the whole thing, which we want for other reasons).